File: nsga2/evaluate.js
module.exports = (i, data) =>{
	let t1 = 0;
	let t2 = 0;
	data.forEach(function(fitness){
		t1 += fitness.obj1;
		t2 += fitness.obj2;
	})
	console.log('Generation #'+i, 'obj1: '+t1/data.length, 'obj2: '+t2/data.length);
}

File: cpp/evaluate.cpp
using namespace rapidjson;
using namespace std;

void evaluate(short i, vector<Solution*>& population){
	double obj1 = 0;
	double obj2 = 0;
	for(short j = 0; j < population.size(); j++){
		obj1 += population[j]->objective1;
		obj2 += population[j]->objective2;
	}
	cout << "Generation #" << i+1 << " obj1: " << obj1 / POPULATION << " obj2: " << obj2 / POPULATION << endl;
}
